src/go-slang/lib/waitgroup.ts

```diff
@@ -0,0 +1,21 @@
+import { WORD_SIZE } from "./heap/config"
+
+export class WaitGroup {
+  protected memory: DataView
+
+  toString(): string { return `WaitGroup { count: ${this.getCount()} }` }
+
+  constructor(memory: DataView) { this.memory = memory }
+
+  protected getCountAddr(): number { return WORD_SIZE }
+
+  protected getCount(): number { return this.memory.getFloat64(this.getCountAddr()) }
+
+  protected setCount(value: number): void { this.memory.setFloat64(this.getCountAddr(), value) }
+
+  public add(n: number): void { this.setCount(this.getCount() + n) }
+
+  public done(): void { this.setCount(this.getCount() - 1) }
```

Comment:
NOTE: I might need to prevent count from going below 0.

Reply:
According to the specs, it panics when counter goes below 0. We could have the ECE raises a runtime error

Ref: https://pkg.go.dev/sync#WaitGroup.Add

Reply:
Good idea.